dto.c

```diff
@@ -790,6 +863,14 @@ static int dsa_init_from_accfg(void)
 		if (i != num_wqs)
 			continue;
 
+		struct dto_device* dev = NULL;
+		uint8_t dev_numa_node = 0;
+
+		if (is_numa_aware) {
+			dev = calloc(1, sizeof(struct dto_device));
```

Comment:
What is there are multiple DSA devices from the same NUMA node? You are storing only one dto_device per numa node in the devices[] array. there will be memory leak here.